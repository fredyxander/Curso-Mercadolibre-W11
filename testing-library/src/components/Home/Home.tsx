export const Home = ()=>{
  return(
    <div>
      <h1 className="home-title" data-testid="home-title">Bienvenida!</h1>
      <h3>Bootcamp 11</h3>
      <p className="content-info">Lorem ipsum dolor sit amet consectetur adipisicing elit. Harum labore cupiditate libero beatae repellendus similique omnis excepturi ullam officiis odit.\</p>
      <button>Ver mas</button>
      <img src=""/>
      <a href="">Ver productos</a>
    </div>
  )
}