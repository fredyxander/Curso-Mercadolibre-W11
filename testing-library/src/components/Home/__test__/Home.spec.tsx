import {render} from "@testing-library/react";
import "@testing-library/jest-dom";
import { Home } from "../Home";

describe("Pruebas para el componente Home", ()=>{
  
  it("Renderizado inicial de Home", ()=>{
    const {container} = render(<Home/>);
    expect(container).toMatchSnapshot();
  });

  it("Se renderizan los datos correctamente",()=>{
    const {getByTestId} =  render(<Home/>);
    const title = getByTestId("home-title");
    expect(title).toHaveTextContent("Bienvenida!");
  });

});