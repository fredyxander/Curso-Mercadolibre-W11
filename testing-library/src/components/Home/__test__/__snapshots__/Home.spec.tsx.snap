// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente Home Renderizado inicial de Home 1`] = `
<div>
  <div>
    <h1
      class="home-title"
      data-testid="home-title"
    >
      Bienvenida!
    </h1>
    <h3>
      Bootcamp 11
    </h3>
    <p
      class="content-info"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Harum labore cupiditate libero beatae repellendus similique omnis excepturi ullam officiis odit.\\
    </p>
    <button>
      Ver mas
    </button>
    <img />
    <a
      href=""
    >
      Ver productos
    </a>
  </div>
</div>
`;
